design-library/src/components/BccProgress/BccProgress.vue

```diff
@@ -36,37 +33,52 @@ const percentageSafe = computed(() => {
   return percentage.value;
 });
 
-const height = computed(() => {
+const progressClass = computed(() => {
+  if (props.progressClass) return props.progressClass;
   return {
+    "bcc-progress": true,
     "h-1": props.size === "sm",
     "h-1.5": props.size === "base",
     "h-2": props.size === "lg",
   };
 });
+
+const barClass = computed(() => {
+  if (props.barClass) return props.barClass;
+  if (
+    typeof progressClass.value === "object" &&
+    Object.keys(progressClass.value).includes("bcc-progress")
+  ) {
+    const clone: any = { ...progressClass.value };
+    delete clone["bcc-progress"];
+    return {
+      ...clone,
+      "bcc-progress-bar": true,
+    };
+  }
+  return progressClass.value;
+});
```

Comment:
I'm a bit ambivalent towards this. On the one hand it's a nice way of handing control to the consumer of the library. On the other hand, an explicit goal of this library is to have a limited set of options, so developers are forced to work within the constraints of the design system (perhaps this goal should be better articulated somewhere). I personally feel this makes the code a bit less readable as well.

I'd be inclined to suggest to not add these class props. Instead, people can override the classes that are already there (like `bcc-progress`) in the normal CSS way. The Event app (my.event.bcc.no) does this to add dark mode support for example.

What do you think, do you agree with the above reasoning? Maybe we should have a broader discussion about this?

Reply:
Agreed, simplified it now 👍🏼